metadata/src/main/java/org/apache/kafka/controller/PeriodicTask.java

```diff
@@ -54,37 +58,44 @@ class PeriodicTask {
 
     PeriodicTask(
         String name,
-        Supplier<ControllerResult<Boolean>> op,
+        QuorumController.ControllerWriteOperation<Boolean> writeOp,
         long periodNs,
         EnumSet<PeriodicTaskFlag> flags
     ) {
-        this.name = name;
-        this.op = op;
-        this.immediatePeriodNs = DEFAULT_IMMEDIATE_PERIOD_NS;
-        this.periodNs = periodNs;
-        this.flags = flags;
+        this(name, writeOp, periodNs, flags, DEFAULT_IMMEDIATE_PERIOD_NS);
     }
 
     PeriodicTask(
         String name,
-        Supplier<ControllerResult<Boolean>> op,
+        QuorumController.ControllerWriteOperation<Boolean> writeOp,
```

Comment:
I think it would be better to have a separate callback like a `Consumer<Long>` that we can pass in here. Making people implement `QuorumController.ControllerWriteOperation` feels a bit messy since that's an inner class of `QuorumController`. It's also a bit confusing (to me, at least)